library/alloc/src/collections/vec_deque/into_iter.rs

```diff
@@ -55,3 +67,17 @@ impl<T> ExactSizeIterator for IntoIter<T> {
 
 #[stable(feature = "fused", since = "1.26.0")]
 impl<T> FusedIterator for IntoIter<T> {}
+
+#[unstable(feature = "trusted_len", issue = "37572")]
+unsafe impl<T> TrustedLen for IntoIter<T> {}
+
+#[doc(hidden)]
+#[unstable(feature = "trusted_random_access", issue = "none")]
+unsafe impl<T> TrustedRandomAccess for IntoIter<T>
+where
+    T: Copy,
```

Comment:
This should probably get an explanation why only for `Copy` (because of the `Drop` impl), same for the others

Reply:
Thanks, that made me remember that we don't need `T: Copy` on the borrowing iterators.